### PR TITLE
gh-1178 save generated key

### PR DESCRIPTION
**Pull request by sche**

### CODE REVIEW

File: Multisig/Logic/Models/PrivateKey/PrivateKey.swift [hunk 1]
```diff
@@ -19,26 +20,36 @@ struct PrivateKey {
...
     }
 
-    var data: Data {
+    struct PKData: Codable {
+        let key: Data
+        let mnemonic: String?
+    }
+
+    /// needed for PrivateKeyController.migrateLegacySigningKey()
+    var keyData: Data {
```
Comment (DmitryBespalov) [inline Multisig/Logic/Models/PrivateKey/PrivateKey.swift:29]: maybe calling it `legacyKeyData` would be better

File: Multisig/UI/Settings/OwnerKeyManagement/GenerateOwnerKey/OnboardingGenerateKeyViewController.swift [hunk 2]
```diff
@@ -43,7 +52,30 @@ class OnboardingGenerateKeyViewController: UITableViewController {
...
+    @objc private func didTapNextButton(_ sender: Any) {
+        let vc = EnterAddressNameViewController()
+        vc.actionTitle = "Save"
+        vc.descriptionText = "Choose a name for the owner key. The name is only stored locally and will not be shared with Gnosis or any third parties."
+        vc.screenTitle = "Enter Key Name"
+        vc.trackingEvent = .enterKeyName
+        vc.placeholder = "Enter name"
+        vc.address = privateKey.address
+        vc.completion = { [unowned self, vc] name in
+            guard PrivateKeyController.importKey(privateKey, name: name, isDrivedFromSeedPhrase: true) else { return }
```
Comment (DmitryBespalov) [inline Multisig/UI/Settings/OwnerKeyManagement/GenerateOwnerKey/OnboardingGenerateKeyViewController.swift:64]: I'm a bit confused - shouldn't we go directly to the key details after entering key name?

From the task description:
```
When pressed,
will save the seed phrase securely and the private key
Updates tracking user properties and triggers tracking events
Closes the flow and opens the key details
Shows snackbar
The key successfully created. Add it to a Safe on desktop and then restart the mobile app.
```

**Comment by DmitryBespalov** [comment]
@sche ok, as we agreed, please address the comments in another pr 👍

**Review by DmitryBespalov — APPROVED** [review]

File: Multisig/Logic/Models/PrivateKey/PrivateKey.swift (hunk 1 above)
Comment (sche) [reply Multisig/Logic/Models/PrivateKey/PrivateKey.swift:29]: ok

**Pull request merged**